continuous-integration/readme.md

```diff
@@ -1,3 +1,227 @@
+
+In the spirit of transparency and embracing frequent communication across a dev crew, we encourage developers to commit code on a daily cadence. This approach provides visibility to feature progress and accelerates pair programming across the team. Here are some principles to consider:
+
+- [ ] **Everyone commits to the git repository each day**
+  - End of day checked-in code should contain unit tests at the minimum.
+  - Run the build locally before checking in to avoid CI pipeline failure saturation. You should verify what caused the error, and try to solve it as soon as possible instead of committing your code. We encourage developers to follow a [lean SDLC principles](https://leankit.com/learn/lean/principles-of-lean-development/).
+  - Isolate work into small chunks which ties directly to business value and refactor incrementally.
+
+## Isolated Environments
+
+One of the key goals of build validation is to isolate and identify failures in non-production environments and minimize any disruption to live production traffic. Our E2E automated tests should run in an environment which mimics our production environment(as much as possible). This includes consistent software versions, OS, test data volume simulations, etc.
+
+- [ ] **Test in a clone of production**
+  - The production environment should be duplicated into another environment(QA and/or Pre-Prod) at a minimum.
+- [ ] **Pull request update triggers dev integration release**
+  - New commits related to a pull request should trigger a build / release into a dev integration environment. The production environment should be fully isolated from this process.
+- [ ] **Promote infrastructure changes across fixed environments**
+  - Infrastructure as code changes should be tested in devint and promoted to all non-production environment(s) then migrated to production with zero downtime to ultimately minimize disruption(s) to our user base.
+- [ ] **Environment releases are triggered on merges into master**
+  - Code changes merged into the master branch should auto trigger a new release candidate for all cloud environments.
+
+## Developer Access to Latest Release Artifacts
+
+Our devops workflow should enable developers to get, install and run the latest system executable. Release executable(s) should be auto generated as part of our CI/CD pipeline(s). This approach provides nice audibility which ties each git commit to a release artifact and empowers developers to recreate system state from any point in time.
+
+- [ ] **Developers can access latest executable**
+  - Latest system executable is available for all developers on the team. There should be a well-known place where developers can reference the release artifact.
+
+- [ ] **Release artifact is published for each git commit**
```

Comment:
each commit in every branch?

Reply:
each commit for an open pull request. Good catch